experimental/CliffordOrders/src/CliffordOrders.jl

```diff
@@ -0,0 +1,1056 @@
+"""
+function even_coefficients(x::ZZCliffordOrderElem)
+  if isdefined(x, :even_coeffs)
+    return x.even_coeffs
+  end
+  _set_even_odd_coefficients!(x)
+  return x.even_coeffs
+end
+
+@doc raw"""
+    odd_coefficients(x::ZZCliffordOrderElem) -> Vector{QQFieldElem}
+
+Return the coefficient vector of $x$ with respect to the
+canonical basis of its parent Clifford order,
+but all coefficients corresponding to basis elements
+with even grading are set to zero. This also updates
+the field `x.odd_coeffs`.
+"""
+function odd_coefficients(x::ZZCliffordOrderElem)
+  if isdefined(x, :odd_coeffs)
+    return x.odd_coeffs
+  end
+  _set_even_odd_coefficients!(x)
+  return x.odd_coeffs
+end
+
+#############################################################
+#
+#  String I/O
+#
+#############################################################
+
+##### Order #####
+function Base.show(io::IO, ::MIME"text/plain", C::CliffordOrder)
+  io = pretty(io)
+  print(io, "Clifford order of even lattice over $(base_ring(C)) with Gram matrix\n")
+  print(io, Indent())
+  show(io, "text/plain", gram_matrix(C))
+  print(io, Dedent(), "\nand coefficient ideals of the lattice\n")
+  print(io, Indent())
+  show(io, "text/plain", _coefficient_ideals_of_lattice(lattice(C)))
+  print(io, Dedent())
+end
+
+Base.show(io::IO, C::CliffordOrder) = print(io, "Clifford order over $(base_ring(C))")
+
+function Base.show(io::IO, ::MIME"text/plain", C::ZZCliffordOrder)
+  io = pretty(io)
+  print(io, "Clifford order of even integer lattice with Gram matrix\n")
+  print(io, Indent())
+  show(io, "text/plain", gram_matrix(C))
+  print(io, Dedent())
+end
+
+Base.show(io::IO, C::Union{ZZCliffordOrder, CliffordOrder}) = print(io, "Clifford order over $(base_ring(C))")
+
+### Elements ###
+function Base.show(io::IO, x::Union{ZZCliffordOrderElem, CliffordOrderElem})
+  print(io, "[")
+  foreach(y -> print(io, "$y "), coefficients(x)[1:(end - 1)])
+  print(io, "$(coefficients(x)[end])]")
+end
+
+################################################################################
+#
+#  basic functionality
+#
+################################################################################
+
+@doc raw"""
+    zero(C::CliffordOrder) -> CliffordOrderElem
+
+Return the additive identity of the Clifford order $C$.
+"""
+zero(C::CliffordOrder) = C()
+
+@doc raw"""
+    one(C::CliffordOrder) -> CliffordOrderElem
+
+Return the multiplicative identity of the Clifford order $C$.
+"""
+function one(C::CliffordOrder)
+  v = fill(base_ring(ambient_algebra(C))(), rank(C))
+  v[1] = base_ring(ambient_algebra(C))(1)
+  return C(v)
+end
+
+@doc raw"""
+    pseudo_basis(C::CliffordOrder, i::Int) -> PMat{NumFieldElem, NumFieldOrderFractionalIdeal}
+
+Return the $i$-th canonical pseudo-basis element of $C$.
+"""
+function pseudo_basis(C::CliffordOrder, i::Int)
+  ba = base_ring(ambient_algebra(C))
+  res = zero_matrix(ba, 1, rank(C))
+  res[1, i] = ba(1)
+  return pseudo_matrix(res, [coefficient_ideals(C)[i]])
```

Comment:
Hm, shouldn't a pseudo-basis consist of pairs of elements of the algebra and coefficient ideals? At least this is how this terminology is used everywhere else in Oscar.

Reply:
Changed in this [commit](https://github.com/oscar-system/Oscar.jl/pull/4370/commits/d4b78c05016fbec7d6ab36f6b18b4b8bec2d084d)